### DISCUSSION TITLE
Run the game Summertime Saga with nix-alien

### DISCUSSION THREAD

**Discussion by UlyssesZh** [thiagokokada/nix-alien#101] [Q&A]
[Summertime Saga](https://summertimesaga.com/) is a free-to-download game. There is a launcher script, which sets `LD_LIBRARY_PATH` and runs a binary program. I tried to remove the `LD_LIBRARY_PATH` setting and replace the binary program launching with `nix-alien`. Then, when it runs, it immediately throws this error and exit:
```
Traceback (most recent call last):
  File "/home/ulysses/apps/games/SummertimeSaga/SummertimeSaga-0-20-15-pc/SummertimeSaga.py", line 198, in <module>
    main()
  File "/home/ulysses/apps/games/SummertimeSaga/SummertimeSaga-0-20-15-pc/SummertimeSaga.py", line 195, in main
    renpy.bootstrap.bootstrap(renpy_base)
  File "/home/ulysses/apps/games/SummertimeSaga/SummertimeSaga-0-20-15-pc/renpy/bootstrap.py", line 273, in bootstrap
    renpy.display.presplash.start(basedir, gamedir)
  File "/home/ulysses/apps/games/SummertimeSaga/SummertimeSaga-0-20-15-pc/renpy/display/presplash.py", line 64, in start
    pygame_sdl2.display.init()
  File "src/pygame_sdl2/display.pyx", line 96, in pygame_sdl2.display.init
pygame_sdl2.error.error: No available video device
```
I then tried to set the env var `SDL_VIDEODRIVER` to various different things, and still couldn't fix it.

I think there may be some libraries that I need to manually add to `nix-alien` call, but I am not sure. My other guess is that the so libraries used by this program (provided by the game in a `lib` subdir) may be different from those from nixpkgs, so this issue may be related to #77. Could you please see whether there is a way to make the game run?

PS: The game is NSFW. Run it in private area.

**Comment by thiagokokada** [comment]
> I think there may be some libraries that I need to manually add to nix-alien call, but I am not sure.

Please take a look at https://github.com/thiagokokada/nix-alien#limitations for how to manually add libraries.

**Reply by UlyssesZh** [reply]
How to know which libraries to add?

**Reply by thiagokokada** [reply]
In this case you don't. The section that I linked in the `README.md` above explains that it is impossible for `nix-alien` to know which libraries it needs to add because you are in a Python script, not a binary (so it is basically the same case as `dlopen` explained in the `README.md`, where Python is probably calling `ctypes` or something).

You need to know which libraries to import in your program to make it work, and then use either `-l` or `-p` flags to add them.

**Comment by agvantibo-again** [comment]
If you have Steam installed or you don't mind it, you might want to check out steam-run, a FHS-compliant environment launcher that is built for running games. You can modify it in your system configurations if you need additional packages, too.
